source/uwp/Renderer/lib/RenderedAdaptiveCard.cpp

```diff
@@ -294,22 +290,41 @@ namespace AdaptiveNamespace
             ComPtr<IAdaptiveSubmitAction> submitAction;
             RETURN_IF_FAILED(localActionElement.As(&submitAction));
 
+            ABI::AdaptiveNamespace::AssociatedInputs associatedInputs;
+            submitAction->get_AssociatedInputs(&associatedInputs);
```

Comment:
not sure if there's much we can do about the nomenclature at this point, but is it a bit weird that you `get_AssociatedInputs()` to retrieve an enum value that tells you whether this instance might have associated inputs, but `get_UserInputs()` to actually maybe get the inputs (if present)?

Reply:
Ugh. Yes, that's horrible. Like you say though, i'm not sure what we can do about it now. I think we want to stick with the associated inputs naming for the new schema prop (which may someday support a list of inputs) and the API should match. But it's also probably not worth the breaking change to change UserInputs. Hopefully the fact that associatedInputs matches the property in the schema will help somewhat. Also someday we should have some API docs :)

---
In reply to: [537962522](https://github.com/microsoft/AdaptiveCards/pull/5141#discussion_r537962522) [](ancestors = 537962522)

Reply:
Does the naming have to match the schema property? The getters/setters can say `AssociatedInputIds`, perhaps

Reply:
That seems worse for right now, because you're not getting a list of Ids, you're only getting "none" or "auto". I also think matching makes it clearer because the devs can see the mapping to the schema props.

---
In reply to: [538626638](https://github.com/microsoft/AdaptiveCards/pull/5141#discussion_r538626638) [](ancestors = 538626638)

Reply:
I agree with you @RebeccaAnne -- naming should follow the schema to the furthest extent possible (especially since we don't have solid API docs) :)